src/modules/user/user.service.ts

```diff
@@ -26,15 +26,34 @@ export class UsersService {
       .getOne();
   }
 
+  async getByPhone(phone: string) {
+    return await this.userRepository
+      .createQueryBuilder('users')
+      .where('users.phone = :phone')
+      .setParameter('phone', phone)
+      .getOne();
+  }
+
   async create(signupDto: SignupDto) {
-    const user = await this.getByEmail(signupDto.email);
+    const userEmail = await this.getByEmail(signupDto.email);
```

Comment:
Can we use one DB query to check if email or phone exists?
i.e: if user provide both email and phone,  making only one db query to check existence.